app/Http/Controllers/CarController.php

```diff
@@ -101,10 +110,33 @@ public function update(Request $request, string $id)
             'carTitle' => 'required|string',
             'description' => 'required|string|max:1000',
             'price' => 'required|numeric',
+            'image' => 'nullable|image|mimes:jpeg,png,jpg,gif|max:2048',
         ]);
-        $data['published'] = isset($request->published);
+
+        $car = Car::findOrFail($id);
+
+        if ($request->hasFile('image')) {
+
+            $oldImagePath = public_path('assets/images/' . $car->image);
+
+            if (File::exists($oldImagePath)) {
```

Comment:
this is pretty good, you make the delete safe, do you know if you delete before checking file exists or not, it can cause a fatal error